editor/import/resource_importer_scene.cpp

```diff
@@ -119,6 +119,304 @@ void EditorSceneImporter::_bind_methods() {
+}
+
+Ref<ArrayMesh> EditorSceneImporterMesh::get_mesh() {
+	ERR_FAIL_COND_V(surfaces.size() == 0, Ref<ArrayMesh>());
+
+	if (mesh.is_null()) {
+		mesh.instance();
+		for (int i = 0; i < blend_shapes.size(); i++) {
+			mesh->add_blend_shape(blend_shapes[i]);
+		}
+		mesh->set_blend_shape_mode(blend_shape_mode);
+		for (int i = 0; i < surfaces.size(); i++) {
+			Array bs_data;
+			if (surfaces[i].blend_shape_data.size()) {
+				for (int j = 0; j < surfaces[i].blend_shape_data.size(); j++) {
+					bs_data.push_back(surfaces[i].blend_shape_data[j].arrays);
+				}
+			}
+			Dictionary lods;
+			if (surfaces[i].lods.size()) {
+				for (int j = 0; j < surfaces[i].lods.size(); j++) {
+					lods[surfaces[i].lods[j].distance] = surfaces[i].lods[j].indices;
+				}
+			}
+
+			mesh->add_surface_from_arrays(surfaces[i].primitive, surfaces[i].arrays, bs_data, lods);
+			if (surfaces[i].material.is_valid()) {
+				mesh->surface_set_material(mesh->get_surface_count() - 1, surfaces[i].material);
+			}
+			if (surfaces[i].name != String()) {
+				mesh->surface_set_name(mesh->get_surface_count() - 1, surfaces[i].name);
+			}
+		}
+	}
+
+	return mesh;
+}
+
+void EditorSceneImporterMesh::clear() {
+	surfaces.clear();
+	blend_shapes.clear();
+	mesh.unref();
+}
+
+void EditorSceneImporterMesh::_set_data(const Dictionary &p_data) {
+	clear();
+	if (p_data.has("blend_shape_names")) {
+		blend_shapes = p_data["blend_shape_names"];
+	}
+	if (p_data.has("surfaces")) {
+		Array surface_arr = p_data["surfaces"];
+		for (int i = 0; i < surface_arr.size(); i++) {
+			Dictionary s = surface_arr[i];
+			ERR_CONTINUE(!s.has("primitive"));
+			ERR_CONTINUE(!s.has("arrays"));
+			Mesh::PrimitiveType prim = Mesh::PrimitiveType(int(s["primitive"]));
+			ERR_CONTINUE(prim >= Mesh::PRIMITIVE_MAX);
+			Array arr = s["arrays"];
+			Dictionary lods;
+			String name;
+			if (s.has("name")) {
+				name = s["name"];
+			}
+			if (s.has("lods")) {
+				lods = s["lods"];
+			}
+			Array blend_shapes;
+			if (s.has("blend_shapes")) {
+				blend_shapes = s["blend_shapes"];
+			}
+			Ref<Material> material;
+			if (s.has("material")) {
+				material = s["material"];
+			}
+			add_surface(prim, arr, blend_shapes, lods, material, name);
+		}
+	}
+}
+Dictionary EditorSceneImporterMesh::_get_data() const {
+	Dictionary data;
+	if (blend_shapes.size()) {
+		data["blend_shape_names"] = blend_shapes;
+	}
+	Array surface_arr;
+	for (int i = 0; i < surfaces.size(); i++) {
+		Dictionary d;
+		d["primitive"] = surfaces[i].primitive;
+		d["arrays"] = surfaces[i].arrays;
+		if (surfaces[i].blend_shape_data.size()) {
+			Array bs_data;
+			for (int j = 0; j < surfaces[i].blend_shape_data.size(); j++) {
+				bs_data.push_back(surfaces[i].blend_shape_data[j].arrays);
+			}
+			d["blend_shapes"] = bs_data;
+		}
+		if (surfaces[i].lods.size()) {
+			Dictionary lods;
+			for (int j = 0; j < surfaces[i].lods.size(); j++) {
+				lods[surfaces[i].lods[j].distance] = surfaces[i].lods[j].indices;
```

Comment:
Distance or screen coverage?

Reply:
See 
https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Vendor/MSFT_lod

Reply:
Distance

Reply:
All this will be generated internally in Godot, the extensions in GLTF will be ignored